src/main/java/com/massivecraft/factions/listeners/FactionsEntityListener.java

```diff
@@ -211,7 +204,19 @@ public void onPotionSplashEvent(PotionSplashEvent event) {
         if (thrower instanceof Player) {
             Player player = (Player) thrower;
             FPlayer fPlayer = FPlayers.getInstance().getByPlayer(player);
-            if (badjuju && fPlayer.getFaction().isPeaceful()) {
+            if (fPlayer.getFaction().isPeaceful()) {
+                for (PotionEffect effect : event.getPotion().getEffects()) {
+                    if (effect.getType().equals(PotionEffectType.WEAKNESS)) {
+                        // make sure this weakness potion only affects zombie villagers
+                        for (LivingEntity target : event.getAffectedEntities()) {
+                            if (target.getType() != EntityType.ZOMBIE_VILLAGER) {
+                                event.setIntensity(target, 0);
+                            }
+                        }
+                        return;
```

Comment:
What if a potion has multiple effects? I believe we should be nuking all other effects to?